torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -2355,6 +2367,71 @@ def test_non_cont_tensors(self):
             self.assertEqual(t_view, t_ret)
             self.assertFalse(t_ret.is_contiguous())
 
+    @dist_init
+    def test_callback(self):
```

Comment:
Can we add a few more test cases:

1) Test that the callback can be a torchscript function.
2) Attach python callback to future returned by:
     i) rpc_async within torchscript.
     ii) rpc_async calling torchscript function.
3) Add tests to validate that we receive appropriate exceptions if callback doesn't have appropriate signature.

Reply:
Cannot test calling `_then` in a TorchScript function yet, as that would require making `_then` torchscrptable.